src/styles/index.scss

```diff
@@ -1,3 +1,92 @@
+@use 'sass:map';
+
+$daybackground: #eee;
+$dayheight: 100px;
+$daywidth: 100px;
+$daycount: 31;
+$border-width: 1px;
+$days-of-week: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6,
+);
+$boxheight: auto;
+
 body {
   margin: 0;
+  padding: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: 7 * $daywidth + 6 * $border-width;
+  height: $boxheight;
+  gap: $border-width;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: #{$daybackground};
+    box-sizing: border-box;
+    margin: 0;
+    width: $daywidth;
+    height: $dayheight;
+    border: $border-width solid black;
+    transition-duration: 0.5s;
+    &::before {
+      font-size: 30px;
+      justify-content: center;
+      align-self: center;
+      content: attr(data-day);
+    }
+    &:hover{
+      background-color: #FFBFCB;
+      cursor: pointer;
+      transform: translateY(-20px)
+    }
+  }
+
+  @for $i from 1 through $daycount {
+    .calendar__day--#{$i}::before {
+      content: '#{$i}';
+    }
+  }
+
+  @each $day, $start in $days-of-week {
+    &--start-day-#{$day} {
+      .calendar__day:first-child {
+        $start-day: $day;
+        $margin: map.get($days-of-week, $start-day);
+
+        margin-left: calc(#{$margin} * ((#{$daywidth} + #{$border-width})));
+      }
+    }
+  }
+
+  @each $days in (28, 29, 30, 31) {
+    &.calendar--month-length-#{$days} {
+      @for $i from 1 through $days {
+        .calendar__day--#{$i} {
+          display: flex;
```

Comment:
You don't need to generate common styles in the cycle for every element

Reply:
Not fixed

Reply:
I tried removing the display: flex , but it won't pass the stylelint requirements, shall I leave it as it is or redo the loop?